### PR TITLE
validate x-version in validator

### PR DESCRIPTION
**Pull request by chongr**

### CODE REVIEW

File: validator.go [hunk 1]
```diff
@@ -84,6 +85,16 @@ func NewMessageValidatorFromBytes(schemaFile []byte) (IMessageValidator, error)
 			compiler.Draft = jsonschema.Draft4
 
 			schemaURL := fmt.Sprintf("%s/schemas/%s/%s", validator.schemaID, schemaName, version)
+			schemaJSONDecoded := map[string]interface{}{}
+			err = json.Unmarshal(schemaByte, &schemaJSONDecoded)
```
Comment (chongr) [inline validator.go]: I didn't see anything in `compiler.Compile` to add fields like `x-versions` into the result jsonschema struct, so it seemed like decoding the JSON was the best way to pick out the versions

Comment (maroux) [reply validator.go]: Yeah this library ignores `x-` tags 😞

File: validator.go [hunk 2]
```diff
@@ -122,6 +138,7 @@ type messageValidator struct {
 	// Format: (schema name, schema version) => schema
 	//   (parking.created, 3.0) => schema
 	compiledSchemaMap map[string]*jsonschema.Schema
+	schemaVersionsMap map[string]map[string]bool
```
Comment (mngo87) [inline validator.go]: comment explaining structure

File: validator.go (hunk 1 above)
Comment (mngo87) [reply validator.go]: can you add more comments? also I prefer if you break these out into functions like `extractXVersions`

File: validator.go [hunk 3]
```diff
@@ -84,6 +85,16 @@ func NewMessageValidatorFromBytes(schemaFile []byte) (IMessageValidator, error)
 			compiler.Draft = jsonschema.Draft4
 
 			schemaURL := fmt.Sprintf("%s/schemas/%s/%s", validator.schemaID, schemaName, version)
+			schemaJSONDecoded := map[string]interface{}{}
+			err = json.Unmarshal(schemaByte, &schemaJSONDecoded)
+			if err != nil {
+				return nil, err
+			}
+			xVersions, ok := schemaJSONDecoded["x-versions"]
```
Comment (mngo87) [inline validator.go]: constant please

**Comment by chongr** [comment]
I think I addressed all these comments awhile ago? @maroux @mngo87

**Pull request merged**